### PR TITLE
Vote articles

### PR DESCRIPTION
**Pull request by Scodia619**

### CODE REVIEW

File: nc-news-fe/api.js [hunk 1]
```diff
@@ -1,11 +1,37 @@
...
+            console.error('Error fetching data:', error);
+            throw error; // Re-throw the error to handle it where the function is called
         });
 };
 
-export default getAllArticles;
\ No newline at end of file
+export const patchVotesByArticleId = (article_id, inc_votes) => {
+    api.patch(`articles/${article_id}`, {inc_votes: inc_votes}).then(res => {
+        console.log(res)
```
Comment (nc-modou) [inline nc-news-fe/api.js]: Remember to take out console logs like this once you're done debugging

File: nc-news-fe/api.js [hunk 2]
```diff
@@ -1,11 +1,37 @@
...
+        });
+};
+
+export const getArticleById = (article_id) => {
+    return api.get(`/articles/${article_id}`)
+        .then(res => res.data.article)
+        .catch(error => {
+            console.error('Error fetching data:', error);
+            throw error; // Re-throw the error to handle it where the function is called
+        });
```
Comment (nc-modou) [inline nc-news-fe/api.js]: I don't see the point in catching these errors here if you're going to re-throw it anyway and handle elsewhere

**Review by nc-modou — COMMENTED** [review]
Good job on the votes so far, the optimistic rendering works well and votes are persistent. However, the user currently isn't notified if the patch request fails, which would be good user feedback to implement.

And, optionally, it could be good to limit the user to one vote in either direction, but this isn't a dealbreaker.

**Review by nc-poppy-m — APPROVED** [review]
Good job on the alert. Before you merge, make sure to give your 'comment' box an id that matches the label, so that this can be interpreted by a screen reader (check the error in your browser console to see what I mean) - you don't need to resubmit this however. A message such as 'Something went wrong, please try again later' would be more useful to a user than 'AxiosError' but this is up to you.

**Pull request merged**